scipy/special/tests/test_basic.py

```diff
@@ -1987,8 +1987,14 @@ def test_factorial_0d_return_type(self, x, exact):
         assert np.isscalar(special.factorial(x, exact=exact))
 
     def test_factorial2(self):
-        assert_array_almost_equal([105., 384., 945.],
-                                  special.factorial2([7, 8, 9], exact=False))
+        assert_array_almost_equal(
+            [105.0, 384.0, 945.0, 1893.6651594423791, 1.75643648307146, inf, 1 / 3],
+            special.factorial2([7, 8, 9, 9.5, 1.8, -6, -5], exact=False),
+        )
+        assert_array_almost_equal(
```

Comment:
Minor: maybe best to use `assert_allclose` for new code as recommended: https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal.html